snap-engine/srtm3_elevation_model.cc

```diff
@@ -147,15 +176,13 @@ void Srtm3ElevationModel::HostToDeviceThread() {
     copy_var_.notify_all();
 }
 
-void Srtm3ElevationModel::HostToDevice() {
+void Srtm3ElevationModel::TransferToDevice() {
     if (!copy_thread_.joinable()) {
         copy_thread_ = std::thread(&Srtm3ElevationModel::HostToDeviceThread, this);
     }
 }
 
-void Srtm3ElevationModel::DeviceToHost() { CHECK_CUDA_ERR(cudaErrorNotYetImplemented); }
-
-void Srtm3ElevationModel::DeviceFree() {
+void Srtm3ElevationModel::ReleaseFromDevice() {
     if (device_formated_srtm_buffers_info_ != nullptr) {
         LOGI << "Unloading SRTM3 tiles from GPU";
         REPORT_WHEN_CUDA_ERR(cudaFree(this->device_formated_srtm_buffers_info_));
```

Comment:
Due to this exception can be thrown from a destructor, e.g terminate()

Reply:
addressed